### PR TITLE
fix(react-components): toggle legend hides container

### PR DESCRIPTION
**Pull request by ssjagad**
## Overview
When the user toggles the legend property in the property panel, the legend container is also hidden now. 

## Verifying Changes

https://github.com/awslabs/iot-app-kit/assets/145582655/4dadbe50-ca3b-4316-b964-6ecf15827b2e

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/react-components/src/hooks/useECharts/useResizeableEChart.ts [hunk 1]
```diff
@@ -29,23 +30,46 @@ const getChartWidth = (width: number, staticWidth: number) => {
...
+  rightLegend: ChartLegend | undefined
 ) => {
   const { width, height } = size;
   const [leftLegendRef, { width: leftLegendWidth }] = useMeasure<HTMLDivElement>();
-  const [chartWidth, setWidth] = useState(getChartWidth(width, leftLegendWidth));
+  const [chartWidth, setChartWidth] = useState(getChartWidth(width, leftLegendWidth));
 
-  const rightLegendWidth = useMemo(() => width - leftLegendWidth - chartWidth, [width, leftLegendWidth, chartWidth]);
+  const rightLegendWidth = useMemo(() => {
+    if (rightLegend === undefined) {
```
Comment (mnischay) [inline packages/react-components/src/hooks/useECharts/useResizeableEChart.ts:41]: nit: if(!rightLegend)

File: packages/react-components/src/hooks/useECharts/useResizeableEChart.ts [hunk 2]
```diff
@@ -29,23 +30,46 @@ const getChartWidth = (width: number, staticWidth: number) => {
...
+  size: { width: number; height: number },
+  rightLegend: ChartLegend | undefined
 ) => {
   const { width, height } = size;
   const [leftLegendRef, { width: leftLegendWidth }] = useMeasure<HTMLDivElement>();
-  const [chartWidth, setWidth] = useState(getChartWidth(width, leftLegendWidth));
+  const [chartWidth, setChartWidth] = useState(getChartWidth(width, leftLegendWidth));
 
-  const rightLegendWidth = useMemo(() => width - leftLegendWidth - chartWidth, [width, leftLegendWidth, chartWidth]);
+  const rightLegendWidth = useMemo(() => {
```
Comment (mnischay) [inline packages/react-components/src/hooks/useECharts/useResizeableEChart.ts:40]: I dont think you need the memo, you could just calculate on the fly

**Review by mnischay — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/customization/widgets/lineScatterChart/component.tsx [hunk 3]
```diff
@@ -200,8 +203,7 @@ const LineScatterChartWidgetComponent: React.FC<LineScatterChartWidget> = (widge
         thresholds={thresholds}
         significantDigits={significantDigits}
         size={size}
-        legend={legend?.visible == true ? {} : undefined}
-        defaultVisualizationType={mapConnectionStyleToVisualizationType(line?.connectionStyle)}
+        legend={legend?.visible == true ? emptyLegendObject : undefined}
```
Comment (mnischay) [inline packages/dashboard/src/customization/widgets/lineScatterChart/component.tsx:206]: almost never use == , 

also, the legend prop being an object may not make sense, AFAIK, we almost never use those, probably worth just make it a boolean

**Review by mnischay — CHANGES_REQUESTED** [review]
The code changes makes sense except the comment. I think you can help clean up the file / flow a little bit 
by updating the prop to boolean (may be).

File: packages/dashboard/src/customization/widgets/lineScatterChart/component.tsx (hunk 3 above)
Comment (ssjagad) [reply packages/dashboard/src/customization/widgets/lineScatterChart/component.tsx]: the type of legend is an object in line scatter chart properties. It's used as such in convertLegend so keeping it as a object. 

-- I did try making into a boolean and ran into "Property 'position' does not exist on type 'boolean'" on a few files.

File: packages/dashboard/src/customization/widgets/lineScatterChart/component.tsx [hunk 4]
```diff
@@ -200,8 +203,7 @@ const LineScatterChartWidgetComponent: React.FC<LineScatterChartWidget> = (widge
         thresholds={thresholds}
         significantDigits={significantDigits}
         size={size}
-        legend={legend?.visible == true ? {} : undefined}
-        defaultVisualizationType={mapConnectionStyleToVisualizationType(line?.connectionStyle)}
```
Comment (mnischay) [inline packages/dashboard/src/customization/widgets/lineScatterChart/component.tsx:204]: why did you remove this defaultVisualizationType ?

**Review by mnischay — CHANGES_REQUESTED** [review]

**Review by mnischay — COMMENTED** [review]
Also, please make sure that the resizing is not broken

File: packages/dashboard/src/customization/widgets/lineScatterChart/component.tsx (hunk 4 above)
Comment (ssjagad) [reply packages/dashboard/src/customization/widgets/lineScatterChart/component.tsx:204]: Good catch, accidentally removed when resolving merge conflicts.

File: packages/react-components/src/hooks/useECharts/useResizeableEChart.ts [hunk 5]
```diff
@@ -29,22 +33,26 @@ const getChartWidth = (width: number, staticWidth: number) => {
  */
 export const useResizeableEChart = (
   chartRef: MutableRefObject<ECharts | null>,
-  size: { width: number; height: number }
+  size: { width: number; height: number },
```
Comment (mnischay) [inline packages/react-components/src/hooks/useECharts/useResizeableEChart.ts:36]: super minor nit: we have a type for this SizeConfig

**Review by mnischay — APPROVED** [review]
